tests/regression/02-base/46-spawn-global-funptrs.c

```diff
@@ -0,0 +1,33 @@
+#include <pthread.h>
+#include <assert.h>
+
+void foo() {
+  assert(1);
+}
+
+void bar() {
+  assert(1);
+}
+
+void (*funs[2])() = {
+  &foo,
+  &bar
+};
+
+
+extern void magic1();
+extern void magic2(void (*funs[])());
+
+void *t_fun(void *arg) {
+  // just for going to multithreaded mode
+  return NULL;
+}
+
+int main() {
+  pthread_t id;
+  pthread_create(&id, NULL, t_fun, NULL); // enter multithreaded mode
```

Comment:
Should `magic2(funs)` not already go to MT mode?

https://github.com/goblint/analyzer/blob/0df380bc67e1df90533f64d44d62a905f0ca3687/src/util/defaults.ml#L164

Reply:
The caveat with that is it only goes multithreaded when it actually spawns something. Arguably this should happen regardless, but it's hardly useful in the current state anyway where we don't have well-defined semantics and configuration for what unknown functions may and may not do.

But actually you're right that this explicit thread creation is unnecessary for the test case. Calling `magic1` in singlethreaded mode already invalidates `funs` to add unknown to it. I'll simplify the test.